### PR TITLE
Implement reservations by occasion 

### PR DESCRIPTION
**Pull request by basil-boh**
- The tag parameter has been changed to a compulsory field to improve testing and consistency across reservations. 
- Added and edited test cases to ensure it reflects the updated change in addCommand output
- Refactored t/ prefix to o/ to indicate occasions instead of tags

### CODE REVIEW

File: src/main/java/seedu/reserve/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -47,7 +47,7 @@ public class EditCommand extends Command {
             + "[" + PREFIX_EMAIL + "EMAIL] "
             + "[" + PREFIX_NUMBER_OF_DINERS + "NUMBER OF DINERS] "
             + "[" + PREFIX_DATE_TIME + "DATETIME] "
-            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "[" + PREFIX_OCCASION + "TAG]...\n"
```
Comment (feliciahmq) [inline src/main/java/seedu/reserve/logic/commands/EditCommand.java:50]: Tag is no longer used. Replace with `OCCASION`.

File: src/test/java/seedu/reserve/logic/commands/AddCommandTest.java [hunk 2]
```diff
@@ -49,14 +49,25 @@ public void execute_reservationAcceptedByModel_addSuccessful() throws Exception
...
+        Reservation validReservation = new ReservationBuilder().withTags("Birthday").build();;
         AddCommand addCommand = new AddCommand(validReservation);
         ModelStub modelStub = new ModelStubWithReservation(validReservation);
 
         assertThrows(CommandException.class, AddCommand.MESSAGE_DUPLICATE_RESERVATION, ()
                 -> addCommand.execute(modelStub));
     }
 
+    @Test
+    public void execute_missingOccasion_throwsCommandException() {
```
Comment (feliciahmq) [inline src/test/java/seedu/reserve/logic/commands/AddCommandTest.java:61]: Good job adding this test case. It guards against edge cases.

File: src/test/java/seedu/reserve/logic/LogicManagerTest.java [hunk 3]
```diff
@@ -157,20 +168,20 @@ private void assertCommandFailureForExceptionFromStorage(IOException e, String e
             @Override
             public void saveReserveMate(ReadOnlyReserveMate reserveMate, Path filePath)
                     throws IOException {
-                throw e;
+                    throw e;
```
Comment (JohannsenLum) [inline src/test/java/seedu/reserve/logic/LogicManagerTest.java:171]: im not sure but the spacing here looks sus, did it pass the checkStyleTest?

File: docs/UserGuide.md [hunk 4]
```diff
@@ -148,7 +148,7 @@ Format: `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [d/DATE_TIME] [x/NUMBER_OF_DINE
 ![editCommandResult.png](images/editCommandResult.png)
 Examples:
 *  `edit 1 p/91234567 e/johndoe@example.com` Edits the phone number and email address of the 1st reservation to be `91234567` and `johndoe@example.com` respectively.
-*  `edit 2 n/Betsy Crower t/` Edits the name of the 2nd reservation to be `Betsy Crower` and clears all existing tags.
+*  `edit 2 n/Betsy Crower o/` Edits the name of the 2nd reservation to be `Betsy Crower` and clears all existing occasions.
```
Comment (feliciahmq) [inline docs/UserGuide.md:151]: Can we add a test case for this?

File: src/test/java/seedu/reserve/logic/LogicManagerTest.java [hunk 5]
```diff
@@ -157,20 +168,20 @@ private void assertCommandFailureForExceptionFromStorage(IOException e, String e
...
+            new JsonUserPrefsStorage(temporaryFolder.resolve("ExceptionUserPrefs.json"));
         StorageManager storage = new StorageManager(reserveMateStorage, userPrefsStorage);
 
         logic = new LogicManager(model, storage);
 
         // Triggers the saveReserveMate method by executing an add command
         String addCommand = AddCommand.COMMAND_WORD + NAME_DESC_AMY + PHONE_DESC_AMY
-                + EMAIL_DESC_AMY + DINERS_DESC_AMY + DATETIME_DESC_AMY;
-        Reservation expectedReservation = new ReservationBuilder(AMY).withTags().build();
+            + EMAIL_DESC_AMY + DINERS_DESC_AMY + DATETIME_DESC_AMY + TAG_DESC_ANNIVERSARY;
```
Comment (JohannsenLum) [inline src/test/java/seedu/reserve/logic/LogicManagerTest.java:183]: i think the + hereshouold be on the next line!

File: src/main/java/seedu/reserve/logic/parser/AddCommandParser.java [hunk 6]
```diff
@@ -33,23 +33,23 @@ public class AddCommandParser implements Parser<AddCommand> {
...
         argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL,
-                PREFIX_NUMBER_OF_DINERS, PREFIX_DATE_TIME);
+            PREFIX_NUMBER_OF_DINERS, PREFIX_DATE_TIME);
         Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
         Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
         Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
         Diners diners = ParserUtil.parseDiners(argMultimap.getValue(PREFIX_NUMBER_OF_DINERS).get());
         DateTime dateTime = ParserUtil.parseDateTime(argMultimap.getValue(PREFIX_DATE_TIME).get());
-        Set<Tag> tagList = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG));
+        Set<Tag> tagList = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_OCCASION));
```
Comment (feliciahmq) [inline src/main/java/seedu/reserve/logic/parser/AddCommandParser.java:52]: Can we use occassionList instead of tagList?

File: src/main/java/seedu/reserve/logic/parser/AddCommandParser.java [hunk 7]
```diff
@@ -33,23 +33,23 @@ public class AddCommandParser implements Parser<AddCommand> {
...
 
         Reservation reservation = new Reservation(name, phone, email, diners, dateTime, tagList);
```
Comment (feliciahmq) [inline src/main/java/seedu/reserve/logic/parser/AddCommandParser.java:54]: Same for this, can we use occassionList instead of tagList?

**Review by feliciahmq — COMMENTED** [review]
LGTM!

File: src/test/java/seedu/reserve/logic/commands/EditCommandTest.java [hunk 8]
```diff
@@ -65,11 +65,11 @@ public void execute_someFieldsSpecifiedUnfilteredList_success() {
         ReservationBuilder reservationInList = new ReservationBuilder(lastReservation);
         Reservation editedReservation = reservationInList.withName(VALID_NAME_BOB)
                 .withPhone(VALID_PHONE_BOB).withDiners(VALID_DINERS_BOB).withDateTime(VALID_DATETIME_BOB)
-                .withTags(VALID_TAG_HUSBAND).build();
+                .withTags(VALID_OCCASION_BIRTHDAY).build();
 
         EditCommand.EditReservationDescriptor descriptor = new EditReservationDescriptorBuilder()
                 .withName(VALID_NAME_BOB).withDiners(VALID_DINERS_BOB).withDateTime(VALID_DATETIME_BOB)
-                .withPhone(VALID_PHONE_BOB).withTags(VALID_TAG_HUSBAND).build();
+                .withPhone(VALID_PHONE_BOB).withTags(VALID_OCCASION_BIRTHDAY).build();
```
Comment (JohannsenLum) [inline src/test/java/seedu/reserve/logic/commands/EditCommandTest.java:72]: Maybe can add tests for invalid occasion to catch those errors

File: src/test/java/seedu/reserve/logic/parser/AddCommandParserTest.java [hunk 9]
```diff
@@ -154,28 +154,32 @@ public void parse_compulsoryFieldMissing_failure() {
...
-                + DINERS_DESC_BOB + DATETIME_DESC_BOB + TAG_DESC_HUSBAND + TAG_DESC_FRIEND, Email.MESSAGE_CONSTRAINTS);
+                + DINERS_DESC_BOB + DATETIME_DESC_BOB + TAG_DESC_BIRTHDAY
+                + TAG_DESC_ANNIVERSARY, Email.MESSAGE_CONSTRAINTS);
 
 
         // invalid tag
         assertParseFailure(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB
-                + DINERS_DESC_BOB + DATETIME_DESC_BOB + INVALID_TAG_DESC + VALID_TAG_FRIEND, Tag.MESSAGE_CONSTRAINTS);
+                + DINERS_DESC_BOB + DATETIME_DESC_BOB + INVALID_TAG_DESC
+                + VALID_OCCASION_ANNIVERSARY, Tag.MESSAGE_CONSTRAINTS);
```
Comment (JohannsenLum) [inline src/test/java/seedu/reserve/logic/parser/AddCommandParserTest.java:174]: Might want to change the java comment to ocassion

File: src/test/java/seedu/reserve/testutil/ReservationUtil.java [hunk 10]
```diff
@@ -37,7 +37,7 @@ public static String getReservationDetails(Reservation reservation) {
         sb.append(PREFIX_NUMBER_OF_DINERS + reservation.getDiners().value + " ");
         sb.append(PREFIX_DATE_TIME + reservation.getDateTime().toString() + " ");
         reservation.getTags().stream().forEach(
-            s -> sb.append(PREFIX_TAG + s.tagName + " ")
+            s -> sb.append(PREFIX_OCCASION + s.tagName + " ")
```
Comment (JohannsenLum) [inline src/test/java/seedu/reserve/testutil/ReservationUtil.java:40]: might want to change tagName to occasionName

**Review by JohannsenLum — COMMENTED** [review]
Overall Looks good! just some minor naming and checkStyle conventions

File: src/test/java/seedu/reserve/logic/LogicManagerTest.java (hunk 5 above)
Comment (wowwwp) [inline src/test/java/seedu/reserve/logic/LogicManagerTest.java:183]: Maybe change TAG_DESC_ANNIVERSARY to OCCASSION_DESC_ANNIVERSARY for consistency

**Review by wowwwp — COMMENTED** [review]
overall lgtm just minor changes which others have covered

File: src/test/java/seedu/reserve/logic/LogicManagerTest.java (hunk 3 above)
Comment (basil-boh) [reply src/test/java/seedu/reserve/logic/LogicManagerTest.java:171]: wait I thought so too! but it passed the check style so all is good

File: src/test/java/seedu/reserve/logic/parser/AddCommandParserTest.java (hunk 9 above)
Comment (basil-boh) [reply src/test/java/seedu/reserve/logic/parser/AddCommandParserTest.java:174]: resolved

File: src/test/java/seedu/reserve/testutil/ReservationUtil.java (hunk 10 above)
Comment (basil-boh) [reply src/test/java/seedu/reserve/testutil/ReservationUtil.java:40]: resolved

File: src/main/java/seedu/reserve/logic/parser/AddCommandParser.java (hunk 7 above)
Comment (basil-boh) [reply src/main/java/seedu/reserve/logic/parser/AddCommandParser.java:54]: eh thanks for the catch! will change

File: docs/UserGuide.md (hunk 4 above)
Comment (basil-boh) [reply docs/UserGuide.md:151]: oh ya!!!! good spot fr thanks will add it in
